Include/Rover/Range.hpp

```diff
@@ -187,15 +188,15 @@ namespace Rover {
   template<typename TypeFwd>
   typename Range<B, E, G>::Type Range<B, E, G>::round(
       Evaluator& evaluator, TypeFwd&& value) {
+    using namespace std;
     if constexpr(std::is_same_v<G, void>) {
       return std::forward<TypeFwd>(value);
     } else {
       auto granularity = evaluator.evaluate(m_granularity);
-      auto flr = static_cast<Type>(granularity * static_cast<int64_t>(value /
-        granularity));
-      auto ceil = static_cast<Type>(granularity * (static_cast<int64_t>(value /
-        granularity) + 1));
-      if(std::abs(value - flr) <= std::abs(ceil - value)) {
+      auto flr = static_cast<Type>(granularity * floor(value / granularity));
```

Comment:
On the C++ side, if you have Begin = End = Type = int, the argument yields a double; explicit cast back to int is then necessary to bypass compiler warnings for the narrowing conversion.

Reply:
Okay in that case implement a function ```floor(int)``` implemented as the identity function within the Details namespace and include ```using namespace Rover::Details```. ADL will pick up your overload as it is an exact match and use it in that scenario. The goal is to avoid performing explicit casts as much as possible.

Reply:
Well, the problem is _value_ here is already a double - it stems from the uniform real distribution in [0.0, 1.0]. So it is not about the floor function at all. As long as I use the real distribution, at least one cast somewhere in the code will be required.
And one more thing, right now Type depends on Begin and End only. In other words, with Begin=int, End=int, Granularity=double, the generator still yields an int. I am not sure whether it is correct, but this is one more reason for the cast as of now.

Reply:
Yes that is absolutely right, thanks for the correction.